proto/tendermint/liquidity/v1beta1/tx.proto

```diff
@@ -0,0 +1,207 @@
+    (gogoproto.nullable) = false,
+    (gogoproto.moretags) = "yaml:\"deposit_coins\"",
+    (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins",
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "[{\"denom\": \"denomX\", \"amount\": \"1000000\"}, {\"denom\": \"denomY\", \"amount\": \"2000000\"}]",
+      format: "sdk.Coins"
+    }];
+
+}
+
+// MsgDepositWithinBatchResponse defines the Msg/DepositWithinBatch response type.
+message MsgDepositWithinBatchResponse {}
+
+// `MsgWithdrawWithinBatch` defines an `sdk.Msg` type that supports submitting withdraw request to the batch of the liquidity pool
+// Withdraw submit to the batch from the Liquidity pool with the specified `pool_id`, `pool_coin` of the pool
+// this requests are stacked in the batch of the liquidity pool, not immediately processed and
+// processed in the `endblock` at once with other requests.
+//
+// See: https://github.com/tendermint/liquidity/blob/develop/x/liquidity/spec/04_messages.md
+message MsgWithdrawWithinBatch {
+  option (gogoproto.equal)           = false;
+  option (gogoproto.goproto_getters) = false;
+
+  string withdrawer_address = 1 [ (gogoproto.moretags) = "yaml:\"withdrawer_address\"",
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      description: "account address of the origin of this message",
+      example: "\"cosmos1e35y69rhrt7y4yce5l5u73sjnxu0l33wvznyun\"",
+      format: "sdk.AccAddress"
+    }];
+  // id of the target pool
+  uint64 pool_id = 2 [(gogoproto.moretags) = "yaml:\"pool_id\"", (gogoproto.jsontag) = "pool_id",
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "\"1\"",
+      format: "uint64"
+    }];
+  cosmos.base.v1beta1.Coin pool_coin = 3 [
+    (gogoproto.nullable) = false,
+    (gogoproto.moretags) = "yaml:\"pool_coin\"",
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "{\"denom\": \"poolD35A0CC16EE598F90B044CE296A405BA9C381E38837599D96F2F70C2F02A23A4\", \"amount\": \"1000\"}",
+      format: "sdk.Coin"
+    }];
+}
+
+// MsgWithdrawWithinBatchResponse defines the Msg/WithdrawWithinBatch response type.
+message MsgWithdrawWithinBatchResponse {}
+
+// `MsgSwapWithinBatch` defines an sdk.Msg type that supports submitting swap offer request to the batch of the liquidity pool
+// Swap offer submit to the batch to the Liquidity pool with the specified the `pool_id`, `swap_type_id`,
+// `demand_coin_denom` with the coin and the price you're offering
+// and `offer_coin_fee` must half of offer coin amount * current `params.swap_fee_rate` for reservation to pay fees
+// this requests are stacked in the batch of the liquidity pool, not immediately processed and
+// processed in the `endblock` at once with other requests
+// You should request the same each field as the pool
+// Currently, only the default `swap_type_id`1 is available on this version
+// The detailed swap algorithm can be found here.
```

Comment:
```suggestion
// `MsgSwapWithinBatch` defines an sdk.Msg type that submits a swap offer request to the liquidity pool batch. 
// The swap offer is submitted with the specified the `pool_id`, `swap_type_id`,
// `demand_coin_denom` with the coin and the price you're offering.
//  The `offer_coin_fee` must be half of offer coin amount * current `params.swap_fee_rate` to qualify for a reservation to pay fees. 
// This request is added to the pool and executed at the end of the batch (`endblock`).
// You must submit the request using the same fields as the pool.
// Only the default `swap_type_id`1 is supported.
// The detailed swap algorithm is shown here:
```

Reply:
we might need some minor adjustments until we are clear exactly what actions we are describing here

Reply:
Sound good 👍